app/controllers/orders_controller.rb

```diff
@@ -28,4 +28,10 @@ def update
     order = current_user.update_order params
     redirect_to edit_order_path(order)
   end
+
+  private
+
+  def load_edit?
+    !current_user.can_edit_order_at?(@order.organization) || @order.order_submitted? && !current_user.super_admin?
```

Comment:
How about move this logic into `current_user.can_edit_order?(@order)` instead? I feel that would capture this logic better, and remove the need for a private method in the controller (which always makes me a bit nervous, as it tends to feel like it should be in some model instead)